### PR TITLE
MBQL: handle nils in SQL filters more intuitively

### LINKED ISSUE #13332: "Does not contain" and "Is not" filter also removes nulls
When applying a "does not contain" filter, it implicitly also filters away rows with null in that column. 

I think this is SQL behaviour, but not very nice UX.

### PR DESCRIPTION
**Pull request by sbelak**
SQL has somewhat unintuitive semantics where NULLs get propagated to the top filter expressions. For example
`my_field NOT LIKE '%ar'` will not return rows where `my_field` is null. This is not intuitive and as a user almost never what I want. This PR changes our MBQL compiler for SQL dialects to emit an additional `OR my_field IS NULL` in such cases.

Since this is SQL-only behaviour it's implemented in the query processor for SQL only. 

I opted for an end-to-end test as I think we should be testing the result, rather than the query generated which is just an implementation detail.

Fixes #13332

p.s. it would be nice to have a test dataset with nils in it.

### CODE REVIEW

File: src/metabase/driver.clj [hunk 1]
```diff
@@ -615,6 +615,6 @@
 
 (defmulti db-start-of-week
   "Return start of week for given database"
-  {:added "0.37.0" :arlists '([driver database])}
+  {:added "0.37.0" :arlists '([driver])}
```
Comment (sbelak) [inline src/metabase/driver.clj:618]: this is completely unrelated, but while harmless, it was wrong

Comment (robdaemon) [reply src/metabase/driver.clj:618]: shouldn't this be `:arglists` and not `:arlists` ?

**Review by robdaemon — CHANGES_REQUESTED** [review]
arglists vs arlists, but that's about it

**Comment by sbelak** [comment]
Implemented code review suggestions and all tests are now green.

**Review by rlotun — APPROVED** [review]

**Review by robdaemon — APPROVED** [review]

**Pull request merged**